File: modules/agent_deprecated/src/agent/mocks/controllers/inventory_controller.py
from typing import Optional

from fastapi import APIRouter, Query
from pydantic import BaseModel

from agent.mocks.services import inventory_service

router = APIRouter(prefix="/api", tags=["inventory"])


class TransferRequest(BaseModel):
    sku: str
    from_location: str
    to_location: str
    qty: int


@router.get("/inventory")
async def get_inventory(location: Optional[str] = Query(None)):
    return inventory_service.get_stock(location)


@router.get("/inventory/{sku}")
async def get_inventory_by_sku(sku: str):
    return inventory_service.get_stock_by_sku(sku)


@router.post("/inventory/transfer")
async def transfer_inventory(req: TransferRequest):
    return inventory_service.transfer_stock(
        req.sku, req.from_location, req.to_location, req.qty
    )


@router.get("/locations")
async def get_locations():
    return inventory_service.get_locations()

File: modules/agent/src/mocks/controllers/inventoryController.ts
import { Router, Request, Response } from 'express';
import * as inventoryService from '../services/inventoryService.js';

export const inventoryRouter = Router();

interface TransferRequest {
  sku: string;
  from_location: string;
  to_location: string;
  qty: number;
}

inventoryRouter.get('/inventory', (req: Request, res: Response) => {
  const location = req.query.location as string | undefined;
  res.json(inventoryService.getStock(location));
});

inventoryRouter.get('/inventory/:sku', (req: Request, res: Response) => {
  res.json(inventoryService.getStockBySku(req.params.sku));
});

inventoryRouter.post('/inventory/transfer', (req: Request, res: Response) => {
  const body = req.body as TransferRequest;
  res.json(
    inventoryService.transferStock(body.sku, body.from_location, body.to_location, body.qty)
  );
});

inventoryRouter.get('/locations', (_req: Request, res: Response) => {
  res.json(inventoryService.getLocations());
});
